File: A's/py/546A.py
def main() -> None:
    k, n, w = map(int, input().split())
    price: int = 0

    for i in range(1, w + 1):
        price += i * k

    res: int = price - n
    print(0 if res <= 0 else res)

if __name__ == "__main__":
    main()

File: A's/cpp/546A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int k, n, w;
    cin >> k >> n >> w;
    int price {0};

    for (int i = 1; i <= w; i++) {
        price += i * k;
    }

    int res = price - n;
    if (res <= 0) {
        res = 0;
    }
    cout << res << endl;
    
    return 0;
}